### PR TITLE
Fix handling of negative decimal numbers with no leading digits

### LINKED ISSUE #153: Incorrect values are returned for "-.NNN" vs "-0.NNN"
## What Happened

Not putting a zero in front of negative decimal values causes the calculator to return incorrect results:
`-.25^2` returns `16` (incorrect) but `-0.25^2` returns `0.0625` (correct)
`-.25*.25` returns `0.0625` (should be negative) but `-0.25*0.25` returns `-0.0625`

This behavior is consistent for all negative decimal values I tested.

## Expected Behavior

Negative decimal values should not return different results if they don't have a leading zero.

## Platform Information

```
$ lsb_release -a
No LSB modules are available.
Distributor ID:	elementary
Description:	elementary OS 5.1.7 Hera
Release:	5.1.7
Codename:	hera
```

### PR DESCRIPTION
**Pull request by jeremypw**
Fixes #153

Also improved clarity a little and fixed a couple of long lines.

### CODE REVIEW

File: src/Core/Token.vala [hunk 1]
```diff
@@ -33,12 +33,45 @@ namespace PantheonCalculator.Core {
...
+        }
+        public bool is_right_paren () {
+            return token_type == TokenType.P_RIGHT;
+        }
+        public bool is_alpha () {
+            return token_type == TokenType.ALPHA;
+        }
+        public bool is_minus_sign () {
+            return token_type == TokenType.OPERATOR && (content in "-−");
         }
```
Comment (danrabbit) [inline src/Core/Token.vala:75]: I'm really :-1: on this kind of thing. I don't think it makes it any more clear, except maybe for the minus sign :(

But either way, this should probably be a separate PR since it isn't required for solving this issue and makes the PR a lot harder to see what is actually changed

Comment (jeremypw) [reply src/Core/Token.vala:75]: I have to disagree about the clarity - I found the complex logic in the Scanner much easier to read this way and the lines are shorter, helping to locate the bug.  However, I agree it is not essential to this PR.

**Comment by jeremypw** [comment]
Reverted a lot of unnecessary changes.  The critical changes are between Scanner.vala lines 80 - 95 (and the new tests).

File: src/Core/Token.vala (hunk 1 above)
Comment (jeremypw) [reply src/Core/Token.vala]: PR now more focused

**Review by danrabbit — APPROVED** [review]
I can confirm this fixes the issue. Nice inline comments, makes it clearer what's going on. And tests are always nice!
